controllers/usersignup/usersignup_controller_test.go

```diff
@@ -1532,6 +1795,130 @@ func TestUserSignupWithExistingMUROK(t *testing.T) {
 	})
 }
 
+func TestMigrateExistingMURToSpace(t *testing.T) {
+	customTier := newNsTemplateTier("custom", "dev", "stage")
+
+	for testname, testTier := range map[string]*toolchainv1alpha1.NSTemplateTier{
+		"default tier":     baseNSTemplateTier,
+		"non-default tier": customTier,
+	} {
+		t.Run(testname, func(t *testing.T) {
+			member := NewMemberCluster(t, "member1", v1.ConditionTrue)
+			logf.SetLogger(zap.New(zap.UseDevMode(true)))
+			t.Run("MUR exists and still references NSTemplateSet", func(t *testing.T) {
+				userSignup := NewUserSignup()
+				userSignup.Annotations = map[string]string{
+					toolchainv1alpha1.UserSignupUserEmailAnnotationKey: "foo@redhat.com",
+				}
+				userSignup.Labels = map[string]string{
+					toolchainv1alpha1.UserSignupUserEmailHashLabelKey: "fd2addbd8d82f0d2dc088fa122377eaa",
+					"toolchain.dev.openshift.com/approved":            "true",
+				}
+				userSignup.Spec.OriginalSub = "original-sub:foo"
+
+				mur := newMasterUserRecord(userSignup, "member1", testTier, "foo")
+				mur.Labels = map[string]string{toolchainv1alpha1.MasterUserRecordOwnerLabelKey: userSignup.Name}
+				templates := nstemplateSetFromTier(*testTier)
+				ua := toolchainv1alpha1.UserAccountEmbedded{
+					TargetCluster: "member1",
+					SyncIndex:     "123abc", // default value
+					Spec: toolchainv1alpha1.UserAccountSpecEmbedded{
+						UserAccountSpecBase: toolchainv1alpha1.UserAccountSpecBase{
+							NSLimit:       testTier.Name,
+							NSTemplateSet: &templates,
+						},
+					},
+				}
+				// set the user account
+				mur.Spec.UserAccounts = append(mur.Spec.UserAccounts, ua)
+
+				// given
+				r, req, _ := prepareReconcile(t, userSignup.Name, NewGetMemberClusters(member), userSignup, baseNSTemplateTier, customTier, mur)
+
+				InitializeCounters(t, NewToolchainStatus(
+					WithMetric(toolchainv1alpha1.MasterUserRecordsPerDomainMetricKey, toolchainv1alpha1.Metric{
+						string(metrics.External): 1,
+					}),
+					WithMetric(toolchainv1alpha1.UserSignupsPerActivationAndDomainMetricKey, toolchainv1alpha1.Metric{
+						"1,external": 1,
+					}),
+				))
+
+				// when
+				res, err := r.Reconcile(context.TODO(), req)
+
+				// then verify that the Space was created
+				require.NoError(t, err)
+				require.Equal(t, reconcile.Result{}, res)
+
+				actualUserSignup := &toolchainv1alpha1.UserSignup{}
+				err = r.Client.Get(context.TODO(), types.NamespacedName{Name: userSignup.Name, Namespace: req.Namespace}, actualUserSignup)
+				require.NoError(t, err)
+
+				AssertThatUserSignup(t, req.Namespace, actualUserSignup.Name, r.Client).HasLabel(toolchainv1alpha1.UserSignupStateLabelKey, "approved")
+
```

Comment:
and then verify that the UserAccount doesn't have NSTepmlateSet:
```suggestion
				mur = murtest.AssertThatMasterUserRecord(t, mur.Name, r.Client).Exists().Get()
				assert.Nil(t, mur.Spec.UserAccounts[0].Spec.NSTemplateSet)

```